extensions/standard-processors/processors/RetryFlowFile.cpp

```
@@ -0,0 +1,212 @@
+  }
+  return std::make_pair(1, false);
+}
+
+// Returns true on fail on reuse scenario
+bool RetryFlowFile::updateUUIDMarkerAndCheckFailOnReuse(const std::shared_ptr<FlowFileRecord>& flow_file) {
+  const std::string last_retried_by_property_name = retry_attribute_ + ".uuid";
+  const std::string current_processor_uuid = getUUIDStr();
+  std::string last_retried_by_uuid;
+  if (flow_file->getAttribute(last_retried_by_property_name, last_retried_by_uuid)) {
+    if (last_retried_by_uuid != current_processor_uuid) {
+      if (reuse_mode_ == FAIL_ON_REUSE) {
+        logger_->log_error("FlowFile %s was previously retried with the same attribute by a different "
+            "processor (uuid: %s, current uuid: %s). Transfering flowfile to 'failure'...",
+            flow_file->getUUIDStr(), last_retried_by_uuid, current_processor_uuid);
+        return true;
+      }
+      if (reuse_mode_ == WARN_ON_REUSE) {
+        logger_->log_warn("Reusing retry attribute that belongs to different processor. Resetting value to 1.");
+      } else {  // Assuming reuse_mode_ == RESET_REUSE
+        logger_->log_debug("Reusing retry attribute that belongs to different processor. Resetting value to 1.");
+      }
+    }
+  }
+  flow_file->setAttribute(last_retried_by_property_name, getUUIDStr());
+  return false;
+}
+
+bool RetryFlowFile::setRetriesExceededAttributesOnFlowFile(core::ProcessContext* context, const std::shared_ptr<FlowFileRecord>& flow_file) {
+  try {
+    for (const auto& attribute : exceeded_flowfile_attribute_keys) {
+      std::string value;
+      context->getDynamicProperty(attribute, value, flow_file);
+      flow_file->setAttribute(attribute.getName(), value);
+      logger_->log_info("Set attribute '%s' of flow file '%s' with value '%s'", attribute.getName(), flow_file->getUUIDStr(), value);
+    }
+    return true;
+  }
+  catch (const std::exception& e) {
```

What do we expect to throw here?

---

I do not know, probably some exception on trying to set attributes on the flowfile that should not be set, like "UUID". This pattern follows the one set in `UpdateAttribute`. I can potentially extract the functionality to a helper function.

---

I simply don't think we should silently swallow such error (even if we log).
Not being able to set an attribute is a serious core framework issue, I would let that exception propagate.

---

Try-catch block removed.